packages/clarity-js/src/data/extract.ts

```diff
@@ -0,0 +1,139 @@
+import { ExtractSource, Syntax, Type } from "@clarity-types/core";
+import { Event, Setting } from "@clarity-types/data";
+import config from "@src/core/config";
+import encode from "./encode";
+import * as dom from "@src/layout/dom";
+import * as internal from "@src/diagnostic/internal";
+import { Code, Constant, Severity } from "@clarity-types/data";
+
+export let data: {[key: number]: string } = {};
+export let updateKeys: (number | string)[] = [];
+
+let variables : { [key: number]: Syntax[] } = {};
+let selectors : { [key: number]: string } = {};
+
+export function start(): void {
+    let e = config.extract;
+    if (!e) { return; }
+    for (let i = 0; i < e.length; i+=3) {
+        let source = e[i] as ExtractSource;
+        let key = e[i+1] as number;
+        switch (source) {
+            case ExtractSource.Javascript: 
+                let variable = e[i+2] as string;
+                variables[key] = parse(variable);
+                break;
+            case ExtractSource.Cookie: 
+                /*Todo: Add cookie extract logic*/
+                break;
+            case ExtractSource.Text:
+                let match = e[i+2] as string;
+                selectors[key] = match;
+                break;
+            case ExtractSource.Fragment:
+                let fragments =  e[i+2] as string[];
+                dom.setFragments(fragments);
```

Comment:
>dom [](http://example.com/codeflow?start=16&length=3)

Usually we want the dependency to be other way. Where DOM depends on Data. Can we either export fragments here and read it in DOM? We will discuss this offline.